pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1595,9 +1594,9 @@ def __init__(
         if not control_values:
             control_values = "1" * len(control_wires)
 
-        self.hyperparameters["control_wires"] = control_wires
         self.hyperparameters["work_wires"] = work_wires
         self.hyperparameters["control_values"] = control_values
+        self.hyperparameters["n_control_wires"] = len(control_wires)
```

Comment:
I decided to define this variable as a hyperparameter of the gate, but need to adapt the `compute_matrix` method